wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/QuantityValueImpl.java

```diff
@@ -171,36 +192,48 @@ public String toString() {
 		 * @param upperBound
 		 * 		the upper bound of this quantity
 		 * @param unit
-		 * 		the unit of this string, as an IRI to the relevant entity
+		 * 		the unit of this quantity, as an IRI to the relevant entity
 		 */
 		@JsonCreator
 		JacksonInnerQuantity(
 				@JsonProperty("amount") BigDecimal amount,
 				@JsonProperty("lowerBound") BigDecimal lowerBound,
 				@JsonProperty("upperBound") BigDecimal upperBound,
 				@JsonProperty("unit") String unit) {
-			Validate.notNull(amount, "Numeric value cannot be null");
-			Validate.notNull(unit, "Unit cannot be null");
-			Validate.notEmpty(unit, "Unit cannot be empty. Use \"1\" for unit-less quantities.");
+		    this(amount, lowerBound, upperBound, parseUnit(unit));
+		}
 
-			if(lowerBound != null || upperBound != null) {
-				Validate.notNull(lowerBound, "Lower and upper bounds should be null at the same time");
-				Validate.notNull(upperBound, "Lower and upper bounds should be null at the same time");
+        protected static ItemIdValue parseUnit(String unit) {
+		    Validate.notNull(unit, "Unit cannot be null");
+            Validate.notEmpty(unit, "Unit cannot be empty. Use \"1\" for unit-less quantities.");
```

Comment:
I would feel safer to consider both null and `""` (empty string) as `"1"` and canonicalize them as such in case there is slightly wrong JSON out there. It seems fairly unambiguous to me. But it's maybe not a good idea to depart from Wikibase if it rejects these values.

Reply:
I agree, I initially did that and then realized the tests were enforcing the current behaviour so I thought I should better keep it… It does feel like a safe move though.